lib/valueflow.cpp

```diff
@@ -2416,11 +2415,11 @@ static void valueFlowReverse(TokenList* tokenlist,
                              Token* tok,
                              const Token* const varToken,
                              ValueFlow::Value val,
-                             ValueFlow::Value val2,
+                             const ValueFlow::Value& val2,
                              ErrorLogger* /*errorLogger*/,
                              const Settings* settings)
 {
-    std::list<ValueFlow::Value> values = {val};
+    std::list<ValueFlow::Value> values = {std::move(val)};
```

Comment:
If we move the value it will always copy because its a const ref.

Reply:
The parameter is not a const ref - it's passed byvalue.

Clang or clang-tidy (can't remember) will also warn about such cases where a `std::move()` doesn't do anything.

Reply:
I see its `val2` thats passed by const ref.

Reply:
But this is using `val`.